File: Sis104Avance/app/src/main/java/com/example/sis104avance/ejercicios/EjerciciosActivity.java
package com.example.sis104avance.ejercicios;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.sis104avance.MainActivity;
import com.example.sis104avance.R;
import com.example.sis104avance.examenes.ExamenActivity;

public class EjerciciosActivity extends AppCompatActivity {

    Button button_Ecuacion2Grado, button_Estadistica, button_SalirEjercicios;
    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_ejercicios);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        button_Ecuacion2Grado = findViewById(R.id.button_Ecuacion2doGrado);
        button_SalirEjercicios = findViewById(R.id.button_Salir_Ejercicios);
        button_Estadistica = findViewById(R.id.button_Estadistica);

        button_Ecuacion2Grado.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(EjerciciosActivity.this, EcuacionActivity.class);
                startActivity(intent);
            }
        });

        button_Estadistica.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(EjerciciosActivity.this, EstadisticaActivity.class);
                startActivity(intent);
            }
        });

        button_SalirEjercicios.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(EjerciciosActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: trabajoSIS104/app/src/main/java/com/example/trabajosis104/ejercicios/EjerciciosActivity.kt
package com.example.trabajosis104.ejercicios

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.trabajosis104.R

class EjerciciosActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_ejercicios)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val buttonEcuacion = findViewById<Button>(R.id.buttonEcuacion);
        val buttonComplejo = findViewById<Button>(R.id.buttonComplejo);
        val buttonDivRes = findViewById<Button>(R.id.buttonDivRes);
        val buttonRes = findViewById<Button>(R.id.buttonRes);
        val buttonSerie = findViewById<Button>(R.id.buttonSerie);
        val buttonRecursiva = findViewById<Button>(R.id.buttonRecursiva);
        val buttonValidar = findViewById<Button>(R.id.buttonValidar);
        val buttonSalir = findViewById<Button>(R.id.buttonEjercicioSalir);
        val buttonPersona = findViewById<Button>(R.id.buttonPersona);
        val buttonMultiplicacion = findViewById<Button>(R.id.buttonMultiplicacion);

        buttonEcuacion.setOnClickListener {
            val intent = Intent(this, EcuacionActivity::class.java)
            startActivity(intent)
        }

        buttonMultiplicacion.setOnClickListener {
            val intent = Intent(this, MultiplicacionActivity::class.java)
            startActivity(intent)
        }

        buttonValidar.setOnClickListener {
            val intent = Intent(this, StringValidationActivity::class.java)
            startActivity(intent)
        }

        buttonRecursiva.setOnClickListener {
            val intent = Intent(this, SumaActivity::class.java)
            startActivity(intent)
        }

        buttonComplejo.setOnClickListener {
            val intent = Intent(this, ComplejoActivity::class.java)
            startActivity(intent)
        }

        buttonDivRes.setOnClickListener {
            val intent = Intent(this, DivResActivity::class.java)
            startActivity(intent)
        }

        buttonRes.setOnClickListener {
            val intent = Intent(this, SumaMainActivity::class.java)
            startActivity(intent)
        }

        buttonSerie.setOnClickListener {
            val intent = Intent(this, ManualSumActivity::class.java)
            startActivity(intent)
        }

        buttonPersona.setOnClickListener {
            val intent = Intent(this, PersonViewActivity::class.java)
            startActivity(intent)
        }

        buttonSalir.setOnClickListener {
            finish()
        }

    }
}